Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -1483,7 +1509,9 @@ - (void)serverCallback:(BNCServerResponse *)response {
             } else {
                 [BNCPreferenceHelper setSessionParams:NO_STRING_VALUE];
             }
-            
+            if ([BNCPreferenceHelper getNeedAppListCheck]) {
```

Comment:
we only check this in two places, i.e. after install or open. exactly what you suggested!